commands.go

```diff
@@ -26,8 +28,72 @@ func (cmd *SortCommand) Command() *imap.Command {
 	}
 }
 
+func parseSortCrit(fields interface{}) ([]SortCriterion, error) {
+	list, ok := fields.([]interface{})
+	if !ok {
+		return nil, errors.New("List is required as a sort criteria")
+	}
+
+	result := make([]SortCriterion, 0, len(list))
+	reverse := false
+	for _, crit := range list {
+		crit, ok := crit.(string)
+		if !ok {
+			return nil, errors.New("String is required as a sort key")
+		}
+
+		if strings.EqualFold(crit, "REVERSE") {
+			reverse = true
+			continue
+		}
+		switch crit {
+		// TODO: Fix types for constants.
+		case string(SortArrival), SortCc, SortDate, SortFrom, SortSize, SortSubject, SortTo:
+		default:
+			return nil, errors.New("Unknown sort criteria: " + crit)
+		}
+		result = append(result, SortCriterion{
+			Field:   SortField(crit),
+			Reverse: reverse,
+		})
+		reverse = false
+	}
+
+	if reverse {
+		return nil, errors.New("Missing sort key after REVERSE")
+	}
+
+	return result, nil
+}
+
 func (cmd *SortCommand) Parse(fields []interface{}) error {
-	return errors.New("sortthread: not yet implemented")
+	if len(fields) < 3 {
```

Comment:
Is a command with 2 arguments valid?

Reply:
```
BASE.6.4.SORT. SORT Command

   Arguments:  sort program
               charset specification
               searching criteria (one or more)
```

No, unlike SEARCH, charset argument is mandatory.

Reply:
OK